### ISSUE TITLE
Dependency breaks CI builds

### ISSUE THREAD

**Issue by AmazingTurtle** [StatelessStudio/ts-mailgun#36]
Output of my CI

```
 sh: tsc: not found
 npm ERR! code ELIFECYCLE
 npm ERR! syscall spawn
 npm ERR! file sh
 npm ERR! errno ENOENT
 npm ERR! ts-mailgun@0.4.2 postinstall: `tsc --p tsconfig.dist.json`
 npm ERR! spawn ENOENT
 npm ERR! 
 npm ERR! Failed at the ts-mailgun@0.4.2 postinstall script.
 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
 npm ERR! A complete log of this run can be found in:
 npm ERR!     /home/node/.npm/_logs/2020-05-05T22_28_24_240Z-debug.log
```

I am using nestjs (default project configuration with their default dockerfile) and installed ts-mailgun. dev locally worked fine but not on CI.

I should probably mention that it's breaking on `npm ci`. It can't find tsc. When I manually add `tsc` to dependencies, it fails with the following:

```
 > tsc --p tsconfig.dist.json
 error TS5014: Failed to parse file 'tsconfig.dist.json/tsconfig.json': Unexpected token u in JSON at position 0.
```

lmao

**Comment by DrewImm** [comment]
You'll need the typescript compiler installed in your project dev dependencies to build any typescript project:

Should be `npm install --save-dev typescript`

You'll see something like this in your `package.json`
```json
	"devDependencies": {
		...
		"typescript": "^3.7.2"
		...
	},
```

**Comment by beaulac** [comment]
@DrewImm This solution works for `npm` but is still broken when using `yarn`.

It's really unusual for a TS library to publish .ts sources and use `postinstall` to provide JS to consumers of a package. Normally, npm packages are published with included JS + declaration files instead of (or in addition to) TS sources.

Is there a particular reason this library deviates from this convention by handling TS compilation with a consumer-side (`postinstall`) rather than publisher-side (`prepublish`) script?

The [npm docs](https://docs.npmjs.com/misc/scripts#best-practices) themselves recommend against this approach:
> Don’t use install. Use a .gyp file for compilation, and prepublish for anything else. You should almost never have to explicitly set a preinstall or install script. If you are doing this, please consider if there is another option. The only valid use of install or preinstall scripts is for compilation which must be done on the target architecture.

**Comment by DrewImm** [comment]
Thanks for the feedback. Accepting PRs.

@beaulac Any thoughts on if this will be a breaking change? I don't foresee any reason as such.

Will be scheduled for next appropriate release (depending if this is a breaking change)

**Comment by chandrashan** [comment]
same - tried to deploy on google cloud function (typescript) for firebase, same error.

Any suggestions? or need to wait for next release?

**Comment by jorgsiegel** [comment]
I can't deploy on google cloud functions either. I have the following lines in package.json:
`"dependencies": {
.....
 "ts-mailgun": "^0.4.2"
  },
  "devDependencies": {
....
    "typescript": "4.0.2"
}`

Please let me know if this is going to be fixed soon, or otherwise I will need to look for alternatives... :/

**Comment by DrewImm** [comment]
@AmazingTurtle @beaulac @chandrashan @jorgsiegel  Thanks all for the feedback. This has been corrected in v0.4.3
